### PR TITLE
Divider with subheader example update

### PR DESCRIPTION
**Pull request by JoseAlba**
## Description

Improved divider snippet so it shows sub header text below it. Material divider uses divider and sub headers frequently. 

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/divider.dart [hunk 1]
```diff
@@ -36,37 +36,52 @@ import 'theme.dart';
...
+///           // https://material.io/components/dividers#types
+///           Container(
+///             padding: EdgeInsets.only(left: 20),
+///             child: Align(
+///               alignment: Alignment.centerLeft,
+///               child: Text(
+///                 'Subheader',
+///                 style: Theme.of(context).textTheme.bodyText2.copyWith(
+///                     fontSize: 12.0,
+///                     color: Theme.of(context).textTheme.caption.color),
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/divider.dart:68]: nit: trailing comma after caption.color

File: packages/flutter/lib/src/material/divider.dart [hunk 2]
```diff
@@ -36,37 +36,52 @@ import 'theme.dart';
...
+///             thickness: 5,
+///             indent: 20,
+///             endIndent: 0,
+///           ),
+///           // Subheader example from Material spec.
+///           // https://material.io/components/dividers#types
+///           Container(
+///             padding: EdgeInsets.only(left: 20),
+///             child: Align(
+///               alignment: Alignment.centerLeft,
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/divider.dart:63]: AlignmentDirectional.centerStart

**Review by rami-a — APPROVED** [review]

File: packages/flutter/lib/src/material/divider.dart [hunk 3]
```diff
@@ -36,37 +36,52 @@ import 'theme.dart';
...
-///           child: Container(
-///             color: Colors.blue,
-///             child: const Center(
-///               child: Text('Below'),
+///           const Divider(
+///             color: Colors.black,
+///             height: 20,
+///             thickness: 5,
+///             indent: 20,
+///             endIndent: 0,
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/divider.dart:56]: can remove this line since it defaults to 0

**Review by clocksmith — APPROVED** [review]

File: packages/flutter/lib/src/material/divider.dart (hunk 1 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/divider.dart:68]: Done

File: packages/flutter/lib/src/material/divider.dart (hunk 2 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/divider.dart:63]: Done

File: packages/flutter/lib/src/material/divider.dart (hunk 3 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/divider.dart:56]: Done

File: packages/flutter/lib/src/material/divider.dart [hunk 4]
```diff
@@ -36,37 +36,53 @@ import 'theme.dart';
...
+///          // https://material.io/components/dividers#types
+///          Container(
+///            padding: EdgeInsets.only(left: 20),
+///            child: Align(
+///              alignment: AlignmentDirectional.centerStart,
+///              child: Text(
+///                'Subheader',
+///                style: Theme.of(context).textTheme.bodyText2.copyWith(
+///                  fontSize: 12.0,
+///                  color: Theme.of(context).textTheme.caption.color,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/divider.dart:68]: We should be able to look up the color from `Theme.of(context).colorScheme`. Presumably it's the `onSurface` color with some opacity per the spec?

File: packages/flutter/lib/src/material/divider.dart [hunk 5]
```diff
@@ -36,37 +36,53 @@ import 'theme.dart';
...
+///          Expanded(
+///            child: Container(
+///              color: Colors.amber,
+///              child: const Center(
+///                child: Text('Above'),
+///              ),
+///            ),
+///          ),
+///          const Divider(
+///            color: Colors.black,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/divider.dart:52]: Will this work in dark mode? Maybe use `...colorScheme.onSurface`

File: packages/flutter/lib/src/material/divider.dart [hunk 6]
```diff
@@ -36,37 +36,53 @@ import 'theme.dart';
...
+///                style: Theme.of(context).textTheme.bodyText2.copyWith(
+///                  fontSize: 12.0,
+///                  color: Theme.of(context).textTheme.caption.color,
+///                ),
+///                textAlign: TextAlign.start,
+///              ),
+///            ),
+///          ),
+///          Expanded(
+///            child: Container(
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/divider.dart:75]: Similar comments here: lets show developers how to configure this in a way that works with dark mode, by using the color scheme.

File: packages/flutter/lib/src/material/divider.dart (hunk 4 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/divider.dart:68]: Theme.of(context).textTheme.capption,

I checked and using the caption will work for dark mode as well

File: packages/flutter/lib/src/material/divider.dart (hunk 5 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/divider.dart:52]: I removed the color parameter.

File: packages/flutter/lib/src/material/divider.dart (hunk 6 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/divider.dart:75]: Done

File: packages/flutter/lib/src/material/divider.dart [hunk 7]
```diff
@@ -36,37 +36,49 @@ import 'theme.dart';
...
-///               child: Text('Below'),
-///             ),
-///           ),
-///         ),
-///       ],
-///     ),
-///   );
-/// }
+///   Widget build(BuildContext context) {
+///    return Center(
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/divider.dart:40]: The indentation is off here, need an extra space

File: packages/flutter/lib/src/material/divider.dart [hunk 8]
```diff
@@ -36,37 +36,49 @@ import 'theme.dart';
...
+///      child: Column(
+///        children: <Widget>[
+///          Expanded(
+///            child: Container(
+///              color: Colors.amber,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/divider.dart:45]: I'm assuming that this looks OK in dark/light mode, using the default theme (or ColorScheme.light(), and ColorScheme.dark()).

File: packages/flutter/lib/src/material/divider.dart [hunk 9]
```diff
@@ -36,37 +36,49 @@ import 'theme.dart';
...
+///          const Divider(
+///            height: 20,
+///            thickness: 5,
+///            indent: 20,
+///            endIndent: 20,
+///          ),
+///          // Subheader example from Material spec.
+///          // https://material.io/components/dividers#types
+///          Container(
+///            padding: EdgeInsets.only(left: 20),
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/divider.dart:60]: If we're going to use const consistently in an example, then you'll need it here too. Best to check this code with analyze.

**Review by HansMuller — APPROVED** [review]
LGTM with some trivial tweaks.

File: packages/flutter/lib/src/material/divider.dart (hunk 7 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/divider.dart]: done

File: packages/flutter/lib/src/material/divider.dart (hunk 8 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/divider.dart]: It does look good

File: packages/flutter/lib/src/material/divider.dart (hunk 9 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/divider.dart]: done

**Pull request merged**